tests/ext/integrations/curl/distributed_tracing_curl_propagate_asm_event.phpt

```diff
@@ -0,0 +1,40 @@
+--TEST--
+Distributed tracing header tags propagate asm events with curl_exec()
+--SKIPIF--
+<?php if (!extension_loaded('curl')) die('skip: curl extension required'); ?>
+<?php if (!getenv('HTTPBIN_HOSTNAME')) die('skip: HTTPBIN_HOSTNAME env var required'); ?>
+--ENV--
+DD_TRACE_TRACED_INTERNAL_FUNCTIONS=curl_exec
+HTTP_X_DATADOG_TRACE_ID=42
+--FILE--
+<?php
+
+include 'distributed_tracing.inc';
+
+function query_headers() {
+    $port = getenv('HTTPBIN_PORT') ?: '80';
+    $url = 'http://' . getenv('HTTPBIN_HOSTNAME') . ':' . $port .'/headers';
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    $response = curl_exec($ch);
+    curl_close($ch);
+    return dt_decode_headers_from_httpbin($response);
+}
+
+$meta = &DDTrace\root_span()->meta;
+unset($meta["to_remove"]);
+$meta["foo"] = "buzz";
+
+$span = DDTrace\start_span();
+$span->service = "dd";
+
+DDTrace\set_priority_sampling(DD_TRACE_PRIORITY_SAMPLING_USER_KEEP);
+
+
+DDTrace\Testing\emit_asm_event();
+dt_dump_headers_from_httpbin(query_headers(), ['x-datadog-tags']);
+
+?>
+--EXPECT--
+x-datadog-tags: _dd.p.appsec=1,_dd.p.dm=-4
```

Comment:
Could you expand why?

Reply:
I didn't look too close, but the RFC says:

```
In case of one or several ASM events (IAST, WAF, custom events, etc.), the sampling priority of the local traces is set to “force keep” (_sampling_priority_v1:2 (MANUAL_KEEP)). The  propagated decision maker tag _dd.p.dm is also set to -5 (-5 is the ID for ASM). Note that _dd.p.dm must be only set once and never changed, even in downstream services.
```

In this test case, an ASM event is emitted on 1st request, so I guess that _dd.p.dm=-5 should be propagated to the next service, no?